File: find_town_judge.cpp
//Time Complexity O(N)
//Space Complexity O(N)

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
         
        int size = trust.size() ; 
        int arr[n] ;
        for(int i = 0 ; i < n ; i ++){
            arr[i] = 0 ; 
        } 
        for(int i = 0 ; i < size ; i ++){
            int outgoing = trust[i][0] ; 
            int incoming = trust[i][1] ; 
            arr[outgoing - 1] -- ;
            arr[incoming - 1] ++ ;  
        }
        // // for(t : trust){
        // //     int outgoing = t[0] ; 
        // //     int incoming = t[1] ; 
        // //     arr[outgoing - 1]-- ; 
        // //     arr[incoming - 1] ++ ; 
        // // }

        for(int i = 0 ; i < n ; i ++){
            if(arr[i] == n -1){
                return i + 1 ; 
            }
        }

        return -1 ; 
    }
};

File: Find Town Judge.java
// Time Complexity = O(V+E) 
// Space Complexity = O(V)

class Solution {
    public int findJudge(int n, int[][] trust) {
        int [] indegrees = new int[n];
        for(int [] tr : trust){
            int in = tr[1] - 1; 
            int out = tr[0] -1;
            indegrees[in] += 1;
            indegrees[out] -= 1;
        }
        
        for(int i = 0; i<n;i++){
            if(indegrees[i] == n-1){
                return i+1;
            }
        }
        return -1;
        
    }
}